Sources/Fluent/Schema/Builder.swift

```diff
@@ -20,23 +42,47 @@ extension Builder {
     }
 
     public func foreignId<E: Entity>(
+        _ name: String = E.foreignIdKey,
         for entityType: E.Type,
         optional: Bool = false,
-        unique: Bool = false
-    ) {
+        unique: Bool = false,
+        constraint fkCreation: ForeignKeyCreation = .auto
```

Comment:
I understand your point, I myself appreciate the simplicity and clearness of Fluent design and Vapor in general, but I implemented it like that exactly to be very clear about the behaviour. In this implementation you have all the possibilities covered.
The `autoForeignKey` is really useful because create a foreign key constraint is normally what you want to do, but not necessarily in every situation. Sometimes you need a weak connection, controlled by your business logic, so a way to disable the foreign key creation locally for one specific field is necessary. If not, autoForeignKey is almost of no use in real life projects.